src/core/utilities/makeshift_serializer.cc

```diff
@@ -0,0 +1,158 @@
+#include "core/utilities/makeshift_serializer.h"
```

Comment:
this whole file will be torched

Reply:
I'm leaving it in the PR for a few days so you can see it (in case any part of it looks useful), but this will be removed from the PR before merge